jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py

```diff
@@ -900,6 +926,51 @@ async def test_update_application_bad_permission(
     assert result is not None
     assert result["application_name"] == "old-name"
     assert result["application_identifier"] == "old_identifier"
+
+
+@pytest.mark.asyncio
+async def test_update_application_wrong_owner(
+    client,
+    fill_application_data,
+    inject_security_header,
+):
+    """
+    Test that it is not possible to update applications that you don't own.
+
+    This test proves that an application's values are not updated following a PUT request to the
+    /application/<id> endpoint by a user that does not own the application. We show this by asserting that
+    the status code 403 is returned and that the application_data is still the same as before.
+    """
+    inserted_id = await database.execute(
+        query=applications_table.insert(),
+        values=fill_application_data(
+            application_name="old-name",
+            application_identifier="old_identifier",
+            application_owner_email="owner1@org.com",
+            application_description="old description",
+        ),
+    )
+    count = await database.fetch_all("SELECT COUNT(*) FROM applications")
+    assert count[0][0] == 1
+
+    inject_security_header("not-the-owner@forbidden.com", Permissions.APPLICATIONS_EDIT)
+    response = await client.put(
+        f"/jobbergate/applications/{inserted_id}",
+        json=dict(
+            application_name="new_name",
+            application_identifier="new_identifier",
+            application_description="new_description",
+        ),
+    )
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    query = applications_table.select(applications_table.c.id == inserted_id)
+    result = await database.fetch_one(query)
+
+    assert result is not None
+    assert result["application_name"] == "old-name"
+    assert result["application_identifier"] == "old_identifier"
+    assert result["application_description"] == "old description"
```

Comment:
nice. Can you use this same pattern for the test `test_delete_application__fails_if_owner_is_not_verified`?